### PR TITLE
Fix warnings and tests for missing script values

### PR DESCRIPTION
**Pull request by mayya-sharipova**
For main code:
- ScriptModule will NOT produce a deprecation warning anymore,
when `es.scripting.exception_for_missing_value` is not set.
This warning is produced only when missing values in script are used and
is produced only once.

For tests:
- make `es.scripting.exception_for_missing_value` equal to `false`
by default in tests. This behaviour is similar with default
node settings, and with tests running from IDE.

- modify `ScriptDocValuesMissingV6BehaviourTests` to test the default
behaviour in v6, where there are default values for missing values.

- add `ScriptDocValuesMissingV7BehaviourTests` to test the default
behavior in v7 where `es.scripting.exception_for_missing_value`
is set to `true`, and where exceptions are produced for missing
values.

Relates to #29286

### CODE REVIEW

File: docs/painless/painless-getting-started.asciidoc [hunk 1]
```diff
@@ -136,8 +136,9 @@ the document, `doc['field'].value` for this document returns:
 IMPORTANT: Starting in 7.0, `doc['field'].value` throws an exception if
 the field is missing in a document. To enable this behavior now,
 set a {ref}/jvm-options.html[`jvm.option`] 
-`-Des.scripting.exception_for_missing_value=true` on a node. If you do not enable
-this behavior, a deprecation warning is logged on start up.
+`-Des.scripting.exception_for_missing_value=true` on a node. If you do not
+enable this behavior, a one-time deprecation warning is logged when painless
```
Comment (rjernst) [inline docs/painless/painless-getting-started.asciidoc]: For other deprecated uses that potentially happen a lot, we have used `deprecationLog.deprecatedAndMaybeLog`. I think that would be better here. See other deprecated examples inside ScriptDocValues

**Comment by mayya-sharipova** [comment]
@elasticmachine run sample packaging tests

File: docs/painless/painless-getting-started.asciidoc [hunk 2]
```diff
@@ -136,8 +136,9 @@ the document, `doc['field'].value` for this document returns:
 IMPORTANT: Starting in 7.0, `doc['field'].value` throws an exception if
 the field is missing in a document. To enable this behavior now,
 set a {ref}/jvm-options.html[`jvm.option`] 
-`-Des.scripting.exception_for_missing_value=true` on a node. If you do not enable
-this behavior, a deprecation warning is logged on start up.
+`-Des.scripting.exception_for_missing_value=true` on a node. If you do not
+enable this behavior, a one-time deprecation warning is logged for each
```
Comment (rjernst) [inline docs/painless/painless-getting-started.asciidoc]: I think this explanation needs to be updated given the use of deprecatedAndMaybeLog?

File: server/src/test/java/org/elasticsearch/index/fielddata/ScriptDocValuesLongsTests.java [hunk 3]
```diff
@@ -135,12 +128,12 @@ public Void run() {
             }
         }, noPermissionsAcc);
 
-        assertThat(warnings, containsInAnyOrder(
+        assertThat(warnings, hasItems(
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/index/fielddata/ScriptDocValuesLongsTests.java]: This seems to be making the check lenient, so other warnings could also exist. I think this test should be verify all warnings that are emitted?

File: server/src/test/java/org/elasticsearch/index/fielddata/ScriptDocValuesMissingV6BehaviourTests.java [hunk 4]
```diff
@@ -19,39 +19,25 @@
...
-            }
-        }));
-        assertWarnings("Script: returning default values for missing document values is deprecated. " +
-            "Set system property '-Des.scripting.exception_for_missing_value=true' " +
-            "to make behaviour compatible with future major versions.");
+    @Override
+    public boolean enableWarningsCheck() {
+        // missing values has some deprecated warnings
+        // that only are output once which is not easy to check
+        // that's why we disable warnings' checking
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/index/fielddata/ScriptDocValuesMissingV6BehaviourTests.java]: This is no longer true right? And isn't the purpose of this test to check those warnings?

**Review by rjernst — COMMENTED** [review]
A couple more questions

**Comment by mayya-sharipova** [comment]
@rjernst  Ryan, can you please continue to review this PR, when you have time.  I have tried to address your comments.

**Review by rjernst — APPROVED** [review]
Thanks @mayya-sharipova. LGTM

**Pull request merged**